QModManager/API/ModLoading/Internal/QModFactory.cs

```diff
@@ -0,0 +1,228 @@
+                else
+                    mod = FromDll(subDir, dllFiles);
+
+                string folderName = new DirectoryInfo(subDir).Name;
+
+                if (mod == null)
+                {
+                    //Logger.Error($"Unable to set up mod in folder \"{folderName}\"");
+                    earlyErrors.Add(new QMod(folderName), ModStatus.InvalidCoreData);
+                    continue;
+                }
+
+                if (!mod.Enable)
+                {
+                    Logger.Info($"Mod \"{mod.DisplayName}\" is disabled via config, skipping...");
+                    continue;
+                }
+
+                SortResults sortResult = modSorter.Add(mod);
+                switch (sortResult)
+                {
+                    case SortResults.CircularLoadOrder:
+                        earlyErrors.Add(mod, ModStatus.CircularLoadOrder);
+                        break;
+                    case SortResults.CircularDependency:
+                        earlyErrors.Add(mod, ModStatus.CircularDependency);
+                        break;
+                    case SortResults.DuplicateId:
+                        earlyErrors.Add(mod, ModStatus.DuplicateIdDetected);
+                        break;
+                }
+            }
+            List<QMod> modsToLoad = modSorter.CreateFlatList(out PairedList<QMod, ErrorTypes> lateErrors);
+
+            var modList = new PairedList<QMod, ModStatus>(modsToLoad.Count + earlyErrors.Count + lateErrors.Count);
+
+            foreach (QMod mod in modsToLoad)
+            {
+                modList.Add(mod, ModStatus.Success);
+            }
+
+            foreach (Pair<QMod, ModStatus> erroredMod in earlyErrors)
+            {
+                modList.Add(erroredMod.Key, erroredMod.Value);
+            }
+
+            foreach (Pair<QMod, ErrorTypes> erroredMod in lateErrors)
+            {
+                switch (erroredMod.Value)
+                {
+                    case ErrorTypes.DuplicateId:
+                        modList.Add(erroredMod.Key, ModStatus.DuplicateIdDetected);
+                        break;
+                    case ErrorTypes.CircularDependency:
+                        modList.Add(erroredMod.Key, ModStatus.CircularDependency);
+                        break;
+                    case ErrorTypes.CircularLoadOrder:
+                        modList.Add(erroredMod.Key, ModStatus.CircularLoadOrder);
+                        break;
+                    case ErrorTypes.MissingDepency:
+                        modList.Add(erroredMod.Key, ModStatus.MissingDependency);
+                        break;
+                    default:
+                        throw new InvalidOperationException("Invalid error reported by mod sorter");
+                }
+            }
+
+            CheckOldHarmony(modsToLoad);
+
+            return modList;
+        }
+
+        //private static int ReportErrors(List<string> erroredMods)
```

Comment:
What's happening with this?

Reply:
Since I'm moving this towards a single list containing all mods paired with their results,  
I plan to create a specialized logging class that can be provided with this list, at the various stages of patching, to handle logging of errors and ultimately if the Dialog pop-up should be used or not.